modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaEntityBrowserWidgetTest.php

```diff
@@ -282,16 +283,230 @@ public function testFeaturedMediaEntityBrowserWidget(): void {
+    $featured_media_field->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:2]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertFalse($featured_media_field->hasButton('Select images'));
+    $this->assertMediaSelectionHasRemoveButton('Image 2', 'Article Featured media field');
+
+    // Check that 'Image 1' media item is placed before 'Image 2'.
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Fill in the captions and save the node.
+    $this->getSession()->getPage()->fillField('article_field[form][0][featured_media_field][0][caption]', 'Image 1 caption');
+    $this->getSession()->getPage()->fillField('article_field[form][0][featured_media_field][1][caption]', 'Image 2 caption');
+    $this->getSession()->getPage()->pressButton('Create node');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    $this->getSession()->getPage()->pressButton('Edit');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+    $this->getSession()->getPage()->pressButton('Update node');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert the values were saved correctly in the node.
+    $this->assertSession()->pageTextContains('Node with featured media has been created.');
+    $node = $this->drupalGetNodeByTitle('Article node with IEF', TRUE);
+    $expected_values = [
+      '0' => [
+        'target_id' => '1',
+        'caption' => 'Image 1 caption',
+      ],
+      '1' => [
+        'target_id' => '2',
+        'caption' => 'Image 2 caption',
+      ],
+    ];
+    $actual_values = $node->get('featured_media_field')->getValue();
+    $this->assertEquals($expected_values, $actual_values);
+
+    // Edit the node to remove the current first item in the form (Image 2).
+    $node = $this->drupalGetNodeByTitle('Node with featured media', TRUE);
+    $this->drupalGet($node->toUrl('edit-form'));
+    // Open the IEF form of the embedded article.
+    $this->getSession()->getPage()->pressButton('Edit');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    $remove = $this->getSession()->getPage()->find('css', 'input[data-drupal-selector="edit-article-field-form-inline-entity-form-entities-0-form-featured-media-field-0-current-items-0-remove-button"]');
+    $remove->press();
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Assert the image was removed from the field.
+    $this->assertSession()->pageTextNotContains('Image 1');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert the remaining caption should not remain.
```

Comment:
This is confusing. You are asserting that the remaining caption can't be saved when the image is removed.